x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/InferenceResultFieldMapper.java

```diff
@@ -243,6 +246,8 @@ private static void parseFieldInferenceChunkElement(
             if (childMapper instanceof FieldMapper fieldMapper) {
                 parser.nextToken();
                 fieldMapper.parse(childContext);
+                // Reset leaf object after parsing the field
+                context.path().setWithinLeafObject(true);
```

Comment:
This was tricky - sparse vector parser resets `withinLeafObject`

Reply:
We could - but I'd say all field mappers work on the assumption that they're coming from a non leaf object context that they can reset to. This feels more like an exception than an actual rule.